Occasional panics due to concurrent map writes in v1.2.4

We are seeing some panics in prometheus-c5-exporter v1.2.4 due to concurrent map writes:

```
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: 2024/05/13 21:35:29 [ERROR] Failed to connect Get "http://127.0.0.1:9984/c5/proxy/commands?49&1&-v": dial tcp 127.0.0.1:9984: connect: connection refused
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: fatal error: concurrent map read and map write2024/05/13 21:35:29 [INFO] Processed cstagwd [{dc en} {cmpGrp en1} {version 7.5.0.0} {starttime 2024-05-12_04:07:06.108}]
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: goroutine 335442 [running]:
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: main.getGlobalAttrs(...)
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: /home/vcillo/git/prometheus-c5-exporter/main.go:569
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: main.fetchC5StateMetrics({0x777054, 0xa}, {0xc000074690, 0x2f}, 0x0?)
May 13 21:35:29 c5sp6en1-2 prometheus-c5-exporter[23842]: /home/vcillo/git/prometheus-c5-exporter/main.go:594 +0x153
```

Log excerpt to repeated crash of c5-exporter:

[c5exporter.log](https://github.com/communi5/prometheus-c5-exporter/files/15308823/c5exporter.log)

--- a/main.go
+++ b/main.go
@@ -28,8 +28,8 @@ var metricSet *metrics.Set
 
 // Fix missing cmpGrp label when C5 component is shutdown
 var gCmpGrp map[string]MetricAttribute
-var gDc  map[string]MetricAttribute
-var attributesMtx sync.Mutex
+var gDc map[string]MetricAttribute
+var attributesMtx sync.RWMutex
 
 type eventCounter struct {
 	ID    string
@@ -98,7 +98,7 @@ func buildMetricName(prefix string, name string, attrs []MetricAttribute) string
 	if len(attrs) > 0 {
 		labels := string("{")
 		for _, v := range attrs {
-			if (len(v.name) > 0 && len(v.value) > 0) {
+			if len(v.name) > 0 && len(v.value) > 0 {
 				labels += v.name + string(`="`) + v.value + `",`
 			}
 		}
@@ -484,21 +484,21 @@ func processC5StateCounter(prefix string, lines []interface{}, attrs []MetricAtt
 // processC5CounterMetrics will parse a counter output of type EVENT and USAGE for
 // a specific C5 metric.
 //
-// {
-//   "proxyResponseTimeStampAndState:" : "2021-02-25 10:31:48  active",
-//   "clusterInfo" : "DC=1 {Wien} CompGrpId=31 [VAS-1] (masterId=8)",
-//   "counterName" : "BT_CALLS_LIMIT_REACHED",
-//   "counterType" : "EVENT",
-//   "absoluteValue" : 0,
-//   "currentValue" : 0,
-//   "lastValue" : 0,
-//   "tableValues" : [
-//     "name                            absolute   curr   last",
-//     "trunkname1.ipcentrex.internal         0      0      0",
-//     "trunk2.otherprovider.at               0      0      0",
-//   ],
-//   "tableCountInfo" : "curComponentCount2: 14 (10000) "
-// }
+//	{
+//	  "proxyResponseTimeStampAndState:" : "2021-02-25 10:31:48  active",
+//	  "clusterInfo" : "DC=1 {Wien} CompGrpId=31 [VAS-1] (masterId=8)",
+//	  "counterName" : "BT_CALLS_LIMIT_REACHED",
+//	  "counterType" : "EVENT",
+//	  "absoluteValue" : 0,
+//	  "currentValue" : 0,
+//	  "lastValue" : 0,
+//	  "tableValues" : [
+//	    "name                            absolute   curr   last",
+//	    "trunkname1.ipcentrex.internal         0      0      0",
+//	    "trunk2.otherprovider.at               0      0      0",
+//	  ],
+//	  "tableCountInfo" : "curComponentCount2: 14 (10000) "
+//	}
 func processC5CounterMetrics(basePrefix string, data c5CounterResponse, attrs []MetricAttribute) {
 	const event, usage string = "EVENT", "USAGE"
 	prefix := basePrefix + "_" + strings.ToLower(data.CounterName)
@@ -565,6 +565,9 @@ func processBaseMetrics(prefix string, state c5StateResponse, attrs []MetricAttr
 }
 
 func getGlobalAttrs(prefix string) []MetricAttribute {
+	attributesMtx.RLock()
+	defer attributesMtx.RUnlock()
+
 	var tmpAttrs []MetricAttribute
 	tmpAttrs = append(tmpAttrs, gCmpGrp[prefix])
 	tmpAttrs = append(tmpAttrs, gDc[prefix])
@@ -856,7 +859,7 @@ func main() {
 		}
 	})
 
-	if (conf.SIPProxydExtEnabled) {
+	if conf.SIPProxydExtEnabled {
 		// dedicated endpoint for per-service-provider metrics
 		http.HandleFunc("/metrics-extended", func(httpResponse http.ResponseWriter, req *http.Request) {
 			metricSet = metrics.NewSet()

